docs/resource/README.md

```diff
@@ -102,7 +102,56 @@ as specified in the [Resource SDK specification](https://github.com/open-telemet
 | `service.instance.id` | string | The string ID of the service instance. [1] | `my-k8s-pod-deployment-1`; `627cc493-f310-47de-96bd-71410b7dec09` | Recommended |
 | `service.namespace` | string | A namespace for `service.name`. [2] | `Shop` | Recommended |
 
-**[1]:** MUST be unique for each instance of the same `service.namespace,service.name` pair (in other words `service.namespace,service.name,service.instance.id` triplet MUST be globally unique). The ID helps to distinguish instances of the same service that exist at the same time (e.g. instances of a horizontally scaled service). It is preferable for the ID to be persistent and stay the same for the lifetime of the service instance, however it is acceptable that the ID is ephemeral and changes during important lifetime events for the service (e.g. service restarts). If the service has no inherent unique ID that can be used as the value of this attribute it is recommended to generate a random Version 1 or Version 4 RFC 4122 UUID (services aiming for reproducible UUIDs may also use Version 5, see RFC 4122 for more recommendations).
+**[1]:** MUST be unique for each instance of the same `service.namespace,service.name` pair
+(in other words `service.namespace,service.name,service.instance.id` triplet MUST be globally unique).
+The ID helps to distinguish instances of the same service that exist at the same time
+(e.g. instances of a horizontally scaled service). It is preferable for the ID to be persistent
+and stay the same for the lifetime of the service instance, however it is acceptable that
+the ID is ephemeral and changes during important lifetime events for the service
+(e.g. service instance restarts).
+If the instance has no inherent unique ID that can be used as the value of this attribute
+it is recommended to generate a random Version 1 or Version 4
+[RFC 4122](https://www.ietf.org/rfc/rfc4122.txt) UUID. Services aiming for reproducible UUIDs may also
+use Version 5. UUIDs are typically recommended, as we only need an opaque yet reproducible value for
+the purposes of identifying a service instance. Similar to what can be seen in the man page for the
+[`/etc/machine-id`](https://www.freedesktop.org/software/systemd/man/machine-id.html) file, the underlying data, such as pod name and namespace should be treated as
+confidential by this algorithm, being the user's choice to expose it or not via another resource attribute.
+
+When a UUID v5 is generated, the UUID's namespace MUST be set to:
+`${telemetry.sdk.name}.${telemetry.sdk.language}.${service.name}`, so that the same service yields the
```

Comment:
When we say "generated" supposedly this means generated by the SDK at startup time, right?

I am not sure I understand how to satisfy the requirements of UUID v5 generation. From [RFC](https://datatracker.ietf.org/doc/html/rfc4122#section-4.3) the very first step of generations is this:

>Allocate a UUID to use as a "name space ID" for all UUIDs
>      generated from names in that name space; see [Appendix C](https://datatracker.ietf.org/doc/html/rfc4122#appendix-C) for some
      pre-defined values.

The name space itself is not used for generating the UUID v5 directly. First you generate a new UUID and _allocate_ it to that name space. Then that allocated name space is used as the input to hashing algorithm (see subsequent steps of the algorithm in the RFC).

How exactly do you suggest that the SDK satisfies this requirement? It requires someone to take the value of `${telemetry.sdk.name}.${telemetry.sdk.language}.${service.name}`, record it in some database, generate and allocate a corresponding UUID, store the UUID persistently and then use the UUID as the input to the hashing algorithm.

I fail to see how this is doable on the fly in the telemetry SDK.

I think there is some confusion here as to what a name space of a UUID v5 means. The name space is statically defined, for example the DNS is a pre-defined name space and its name space ID is defined in the RFC: 6ba7b810-9dad-11d1-80b4-00c04fd430c8.

You can say that OpenTelemety services are a name space and generate a name space ID for it that should be used by all SDKs. That name space ID must be recorded in Open Telemetry repositories, it cannot be generated at runtime.

So, again I do not see how a dynamic value like `${telemetry.sdk.name}.${telemetry.sdk.language}.${service.name}` can be a name space.

I may be mising something here, please clarify.

Reply:
Thank you for your question, @tigrannajaryan, it was indeed missing a piece: a static value for the seed UUID, similar to the static ones defined in the spec.

What I had in mind was something like this:

```go
	ns := uuid.NewSHA1(uuid.MustParse("some-static-uuid"), []byte("${telemetry.sdk.name}.${telemetry.sdk.language}.${service.name}"))
	fmt.Printf("v5 ns: %v\n", ns)

	id := uuid.NewSHA1(ns, []byte("host.id"))
	fmt.Printf("v5 id: %v\n", id)
```

For the static UUID, I suggest "4d63009a-8d0f-11ee-aad7-4c796ed8e320", an opaque UUID v1 that I generated locally. If the code above looks reasonable to you, let me know and I'll change the OTEP to add both the static UUID and the clarifications needed, along with this code sample.

Reply:
Is it a goal to be UUID v5 RFC compliant? According to the RFC the static UUID (namespace id) should be concatenated with the name and then hashed once (not twice like the code above does). If you want to use ${telemetry.sdk.name}.${telemetry.sdk.language}.${service.name} as an input perhaps you can concatenate it with the host.id name (i.e. make it part of the name)?

I can't tell the significance of doing the generation exactly as the RFC does, perhaps it doesn't matter.

Reply:
So, maybe simply use `${telemetry.sdk.name}.${telemetry.sdk.language}.${service.name}.${host.id}` as the name.

Reply:
The goal is to have it reproducible. Using a static UUID plus a string like you mentioned would achieve that goal as well:

```go
	id := uuid.NewSHA1(uuid.MustParse("4d63009a-8d0f-11ee-aad7-4c796ed8e320"), []byte("${telemetry.sdk.name}.${telemetry.sdk.language}.${service.name}.${host.id}"))
	fmt.Printf("v5 id: %v\n", id)
```

Reply:
Yes, I think that works.

Reply:
Sorry for not updating this PR sooner, the static namespace for the UUID was determined in this thread, a few messages above. I've updated the algorithm and included a code example in Go. The essence of the document is the same, but it should be clearer now how it's supposed to work.

I included the namespace here as well: I originally left it out, as there are quite a few scenarios where it's not available, but given how common it is to run services in namespaced environment like Kubernetes, I decided to explicitly add it here as well.

Reply:
I'm marking this as resolved as I believe this has been addressed fully.